### DISCUSSION TITLE
Issue running Test

### DISCUSSION THREAD

**Discussion by WildRackoon** [alandefreitas/matplotplusplus#294] [Q&A] [answered]
I am unsure if I am running the `generate_examples` visual test correctly, but I also have a runtime issue that looks like a platform error:

- I understand that it only works on MSVC ? If how are the `build` and `cmake-build-debug` directories supposed to be created in the source ? I see those in the `.gitignore`, I won't mention how in-source builds are frowned upon.

- I have this issue running it, am I in the wrong directory ? or is this a platform error ?
```
$ ./matplotplusplus/test/generate_examples/generate_examples 
terminate called after throwing an instance of 'std::regex_error'
  what():  Mismatched '(' and ')' in regular expression
Aborted (core dumped)
```

- Would you mind if I made running the test using `ctest` possible ?

**Comment by alandefreitas** [comment, marked as answer]
These tests are not really compiled and tested on all platforms. They're just there to generate the images in the documentation.

> I understand that it only works on MSVC ?

This is definitely not true. In fact, it's more likely to fail on MSVC and work on GCC/Clang.

> If how are the build and cmake-build-debug directories supposed to be created in the source ?

I don't see how this is related to it working on MSVC. But it's common practice to have the `build`, or `cmake-build-debug`, or `out`, or any common name for the build directory in the root directory. 

This is not considered in-source builds at all. All artifacts are isolated in this `build` directory. It's exactly how most IDEs do that.

> I see those in the `.gitignore`

You don't have to create the directories in `.gitignore`. These directories are only there to ensure devs don't accidentally push these directories.

> I have this issue running it, am I in the wrong directory ? or is this a platform error ?

This is definitely a logic error throwing an exception. I haven't used it in a while because we already generated all examples and don't need to rerun that that often. 

The only way to know why it's failing is to debug it. My hypothesis is the directory and file names have changed in the meanwhile and the regular expressions no longer work.

> Would you mind if I made running the test using ctest possible ?

You can do that locally, but it doesn't sound like a good idea. That's not a good test because it's not a unit test or anything like that. `ctest` is for things that might fail when there's an error elsewhere. This `generate_examples` target is not supposed to fail unless the error is from `generate_examples` itself.

**Reply by WildRackoon** [reply]
Thanks, my bad I did not initially understand how it was with respect to a ` root directory`.
